test/IRGen/async_dynamic_replacement.swift

```diff
@@ -13,17 +13,17 @@ internal func _replacement_number() async -> Int {
 
 // rdar://78284346 - Dynamic replacement should use musttail
 //                   for tail calls from swifttailcc to swifttailcc
-// CHECK-LABEL: define {{.*}} swifttailcc void @"$s25async_dynamic_replacement01_C7_numberSiyYaFTI"
-// CHECK: musttail call swifttailcc void
+// CHECK-LABEL: define {{.*}} swift{{(tail)?}}cc void @"$s25async_dynamic_replacement01_C7_numberSiyYaFTI"
```

Comment:
I'm not super keen on this change. The point of this test is to make sure that we're using `swifttailcc` and `musttail`, so changing it so that it would also accept `swiftcc` seems like the wrong thing to do.

Reply:
Yeah, it can also accept the combination of `swifttailcc` and `tail`, and `swiftcc` and `musttail` while they are apparently wrong... Do you have any idea how to accept only the correct combinations (`swiftcc` and `tail`, or `swifttailcc` and `mustttail`)?

Reply:
Talked with @etcwilde and supporting swifttailcc in wasm backend seems the right direction instead of adjusting checks since all targets except for wasm supports the cc.

So for now, I excluded the change from this PR.

Reply:
As far as only accepting specific combinations goes, I'm not sure that's easy to do with FileCheck. The closest you can get is by using string substitution blocks, but unfortunately FileCheck doesn't let you substitute in the name of a variable in a substitution block, which makes it impossible to write the test.  If it _did_ support that, you could do something like

```
// ... | FileCheck -DSWIFT_tail_MUST=must -DSWIFT__MUST= %s
...
// CHECK-LABEL: define {{.*}} swift[[TAIL:(tail)?]]cc ...
// CHECK: [[SWIFT_[[TAIL]]_MUST]]tail call swift[[TAIL]]cc void
```

but it sadly doesn't allow the nested substitution.